### PR TITLE
[stable/prometheus-consul-exporter] Improved/Updated README and added ServiceMonitor

### PR DESCRIPTION
**Pull request by mogaal**
#### What this PR does / why we need it:

- This PR updates the imageTag to latest v0.5.0 (tested) as default
- Added ServiceMonitor resource in order to pull metrics prometheus-operator automatically
- Updated README which was out of sync referencing a chart doesn't exist. Also added the configuration block to make it more standards

#### Special notes for your reviewer:

I know there are a lof of things to improve in this Chart, especially how the container configuration is passed, or the lack of `CONSUL_HTTP_TOKEN` env to enhance metrics to expose. I am just starting with something small (small improvements) and progressive I'll be improving the Chart adding new features

#### Checklist
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/prometheus-consul-exporter/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    interval: {{ .Values.serviceMonitor.interval }}
+{{- end }}
+{{- if .Values.serviceMonitor.telemetryPath }}
+    path: {{ .Values.serviceMonitor.telemetryPath }}
+{{- end }}
+{{- if .Values.serviceMonitor.timeout }}
+    scrapeTimeout: {{ .Values.serviceMonitor.timeout }}
+{{- end }}
+  jobLabel: {{ include "prometheus-consul-exporter.fullname" . }}
+  namespaceSelector:
```
Comment (vsliouniaev) [inline stable/prometheus-consul-exporter/templates/servicemonitor.yaml:25]: ServiceMonitors select services from the same namespace by default, so you could omit this block.

File: stable/prometheus-consul-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{{- if .Values.serviceMonitor.interval }}
+    interval: {{ .Values.serviceMonitor.interval }}
+{{- end }}
+{{- if .Values.serviceMonitor.telemetryPath }}
+    path: {{ .Values.serviceMonitor.telemetryPath }}
+{{- end }}
+{{- if .Values.serviceMonitor.timeout }}
+    scrapeTimeout: {{ .Values.serviceMonitor.timeout }}
+{{- end }}
+  jobLabel: {{ include "prometheus-consul-exporter.fullname" . }}
```
Comment (vsliouniaev) [inline stable/prometheus-consul-exporter/templates/servicemonitor.yaml]: The job label is a label on the service that will be used. So the block on the service would have to look like
```
labels:
  {{ include "prometheus-consul-exporter.fullname" . }}: my-job-name
```

If no match is found, the service name is used instead. I would drop this entirely or make it configurable from the values file

File: stable/prometheus-consul-exporter/templates/servicemonitor.yaml (hunk 1 above)
Comment (mogaal) [reply stable/prometheus-consul-exporter/templates/servicemonitor.yaml:25]: That is right: it selects services from the same namespace by default. But if you are creating the `ServiceMonitor` in a different namespace ( `.Values.serviceMonitor.namespace`) then you must specify where (namespace) the service to scrape is. It's actually my case: I have all ServiceMonitors resources inside `monitoring` namespace but prometheus-consul-exporter aside consul (`infrastructure` namespace)

File: stable/prometheus-consul-exporter/templates/servicemonitor.yaml (hunk 2 above)
Comment (mogaal) [reply stable/prometheus-consul-exporter/templates/servicemonitor.yaml]: I just removed, let me know if that is alright :-) and sorry for the delay.

**Comment by eskp** [comment]
Is there any movement here?

**Comment by zanhsieh** [comment]
@vsliouniaev 
Please help. Thank you.

**Comment by vsliouniaev** [comment]
@zanhsieh I don't maintain this chart

**Comment by vladh91** [comment]
Hey guys. I was wondering why this PR has been not approved for merge? Having ServiceMonitor template for the exporter could be really nice.(instead of creating your own).

**Pull request closed without merging**